Automattic/wp-calypso
# Add/tiled gallery block

Adds _Tiled Gallery_ block to Gutenberg-extensions.

The block is based on @haszari's previous work in:
https://github.com/Automattic/jetpack/pull/9903 & https://github.com/Automattic/jetpack/pull/9434

Contains some modifications on top of Haszari's work to make this work with latest Gutenberg and with Calypso's linters.

## Testing

1. Checkout this branch in Calypso and in Jetpack checkout `try/gutenberg-externally-built-tile-gallery-block`

2. Link Calypso to Jetpack:
   - in Calypso `npm link`
   - In Jetpack `npm link wp-calypso`

3. Build the block:
    ```
    yarn calypso-sdk gutenberg --editor-script=client/gutenberg/extensions/tiled-gallery/tiled-gallery.jsx --output-dir=./modules/tiled-gallery
    ```

4. Move the block to right place:
   ```
   mv ./modules/tiled-gallery/tiled-gallery-editor.js ./modules/tiled-gallery/block.js
   ```

5. Insert "Tiled gallery" block in Gutenberg and notice the errors but it more or less works.
